cli/cmd/generate_azure.go

```diff
@@ -0,0 +1,680 @@
+package cmd
+
+import (
+	"fmt"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/AlecAivazis/survey/v2"
+	"github.com/imdario/mergo"
+	"github.com/spf13/cobra"
+
+	"github.com/lacework/go-sdk/lwgenerate/azure"
+	"github.com/pkg/errors"
+)
+
+var (
+	// Define question text here so they can be reused in testing
+	QuestionAzureEnableConfig = "Enable Azure Config Integration?"
+	QuestionAzureConfigName   = "Specify Name of the Config"
+	QuestionEnableActivityLog = "Enable Azure Activity Log Integration?"
+	QuestionActivityLogName   = "Specify name of the Activity Log"
+
+	QuestionAzureAnotherAdvancedOpt      = "Configure another advanced integration option"
+	QuestionAzureConfigAdvanced          = "Configure advanced integration options?"
+	QuestionAzureCustomizeOutputLocation = "Provide the location for the output to be written:"
+
+	// Active Directory
+	QuestionEnableAdIntegration = "Create Active Directory Integration?"
+	QuestionADApplicationPass   = "Specify the password of an existing Active Directory application"
+	QuestionADApplicationId     = "Specify the ID of an existing Active Directory application"
+	QuestionADServicePrincpleId = "Specify the Service Principle ID of an existing Active Directory application"
+
+	// Storage Account
+	QuestionUseExistingStorageAccount   = "Enable use of existing Storage Account?"
+	QuestionAzureRegion                 = "Specify the Azure region to be used by Storage Account logging"
+	QuestionStorageAccountName          = "Specify existing Storage Account name"
+	QuestionStorageAccountResourceGroup = "Specify existing Storage Account Resource Group"
+
+	// Subscriptions
+	QuestionEnableAllSubscriptions = "Enable all subscriptions?"
+	QuestionSubscriptionIds        = "Specify list of subscription ids to enable logging"
+
+	// Management Group
+	QuestionEnableManagementGroup = "Enable Management Group level?"
+	QuestionManagementGroupId     = "Specify Management Group ID"
+
+	// Select options
+	AzureAdvancedOptDone       = "Done"
+	AdvancedAdIntegration      = "Configure Lacework integration with an existing Active Directory"
+	AzureExistingStorageAcount = "Configure Existing storage account"
+	AzureSubscriptions         = "Configure Subscriptions"
+	AzureManagmentGroup        = "Configure Management Group"
+	AzureStorageGroup          = "Configure Storage Group"
+	AzureUserIntegrationNames  = "Configure user integration names"
+	AzureAdvancedOptLocation   = "Customize output location"
+
+	GenerateAzureCommandState      = &azure.GenerateAzureTfConfigurationArgs{}
+	GenerateAzureCommandExtraState = &AzureGenerateCommandExtraState{}
+	CachedAzureAssetIacParams      = "iac-azure-generate-params"
+	CachedAzureAssetExtraState     = "iac-azure-extra-state"
+
+	// List of valid Azure Storage locations
+	validStorageLocations = map[string]bool{
```

Comment:
Would be good to get other opinions on this. I think we've intentionally avoided having an exact list of locations for AWS and GCP. Instead, preferring a regex for rough validation. I believe we were wanting to try and avoid maintaining a comprehensive list of locations that could be constantly in flux.

Reply:
I am not sure that there is a regex for this as they are simply a list of locations, the only regex that I think would fit would really allow any combination of words and numbers which would not really give any gain I think